Contributor
zhengruifeng commented on Sep 2, 2016 â€¢
edited by TomDLT
Description
GaussianProcess batch predict fail on py3 due to range(float)
Steps/Code to Reproduce
import numpy as np
import sklearn.datasets
from sklearn import gaussian_process

digits = sklearn.datasets.load_digits()

X = digits.data
y = digits.target

indices = np.arange(X.shape[0])
np.random.shuffle(indices)
X = X[indices]
y = y[indices]
X_train = X[:1000]
y_train = y[:1000]
X_test = X[1000:]
y_test = y[1000:]

gp = gaussian_process.GaussianProcess()
gp.fit(X_train, y_train)

gp.predict(X_test, batch_size=100)
Expected Results
No error is thrown.
Actual Results
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-b96b929502ec> in <module>()
----> 1 gp.predict(X_test, batch_size=100)

/usr/local/lib/python3.5/site-packages/sklearn/gaussian_process/gaussian_process.py in predict(self, X, eval_MSE, batch_size)
    520
    521                 y = np.zeros(n_eval)
--> 522                 for k in range(max(1, n_eval / batch_size)):
    523                     batch_from = k * batch_size
    524                     batch_to = min([(k + 1) * batch_size + 1, n_eval + 1])

TypeError: 'float' object cannot be interpreted as an integer
Versions
scikit-learn (0.17.1)
Python 3.5.2